src/handlers/suggest/handler.js

```diff
@@ -1,90 +1,178 @@
-}
+    const title = request.payload.new
+        ? 'New company suggestion' + (company.name ? ': `' + company.name + '`' : '')
+        : 'Suggested update for company `' + company.slug + '`';
+    const commit_msg =
+        (request.payload.new
+            ? 'Add ' + (company.name ? ': `' + company.name + '`' : '')
+            : 'Update `' + company.slug + '`') + ' (community contribution)';
 
-async function suggest(request, h) {
-    try {
-        const request_body = request.payload;
-
-        const result = await octokit.issues.create({
+    return await octokit
+        .createPullRequest({
             owner: config.suggest.owner,
             repo: config.suggest.repo,
-            title: request_body.new
-                ? 'New company suggestion' + (request_body.data.name ? ': `' + request_body.data.name + '`' : '')
-                : 'Suggested update for company `' + request_body.data.slug + '`',
-            // we create the issue just with the raw submission as body
-            // this is overwritten later on: the issue is added as source and it becomes a diff if this was an update
-            // but in case that second step fails, we have saved the raw data
-            body: JSON.stringify(request_body.data),
-        });
+            base: config.suggest.branch,
+            title,
+            body: 'This suggestion was submitted through the website.',
+            head: 'suggest_' + company.slug + '_' + Date.now(),
+            changes: [
+                {
+                    files,
+                    commit: commit_msg,
+                },
+            ],
+        })
+        .then((pr) => {
+            if (pr?.data) {
+                company.sources.push(pr.data.html_url);
 
-        if (result.status !== 201) {
-            return h.response({ message: 'Failed to create GitHub issue.' }).code(502);
-        }
+                return commitStringFileToPullRequest(
+                    pr.data.head.user.login,
+                    config.suggest.repo,
+                    pr.data,
+                    file_path,
+                    JSON.stringify(company, null, 4) + '\n', // File content
+                    commit_msg
+                )
+                    .then(() =>
+                        octokit.pulls.update({
+                            owner: config.suggest.owner,
+                            repo: config.suggest.repo,
+                            pull_number: pr.data.number,
+                            body: `This suggestion was submitted through the website.
 
-        const issue_url = `https://github.com/${config.suggest.owner}/${config.suggest.repo}/issues/${result.data.number}`;
+**[Edit](https://company-json.netlify.com/#!doc=${encodeURIComponent(JSON.stringify(company))})**`,
+                            maintainer_can_modify: true, // We cannot set this setting through the plugin, but because it is jus a gimmick, we can do it afterwards just as well.
```

Comment:
:P 

```suggestion
                            maintainer_can_modify: true, // We cannot set this setting through the plugin, but because it is just a gimmick, we can do it afterwards just as well.
```